regulation-worker/internal/delete/delete.go

```diff
@@ -3,31 +3,44 @@ package delete
 import (
 	"context"
 	"fmt"
+	"sync"
 
 	"github.com/rudderlabs/rudder-server/regulation-worker/internal/model"
 )
 
-type deleter interface {
-	Delete(ctx context.Context, job model.Job, dest model.Destination) (model.JobStatus, error)
+//go:generate mockgen -source=delete.go -destination=mock_delete_test.go -package=delete github.com/rudderlabs/rudder-server/regulation-worker/internal/delete
+type deleteManager interface {
+	Delete(ctx context.Context, job model.Job, destConfig map[string]interface{}, destName string) model.JobStatus
+	GetSupportedDestinations() []string
 }
-type Deleter struct {
-	API   deleter
-	Batch deleter
+
+type DeleteRouter struct {
```

Comment:
to avoid repetition `delete.DeleteRouter` -> `delete.Router`

```suggestion
type Router struct {
```